src/contracts/Harvester.sol

```diff
@@ -0,0 +1,267 @@
+import {IHarvestable, IMagpieRouter, IOracle} from "./Interfaces.sol";
+
+/**
+ * @title Collects rewards from strategies and swaps them for ARM liquidity tokens.
+ * @author Origin Protocol Inc
+ */
+contract Harvester is Initializable, OwnableOperable {
+    using SafeERC20 for IERC20;
+
+    enum SwapPlatform {
+        Magpie
+    }
+
+    /// @notice All reward tokens are swapped to the ARM's liquidity asset.
+    address public immutable liquidityAsset;
+    /// @notice The address of the Magpie router that performs swaps
+    address public immutable magpieRouter;
+
+    /// @notice Mapping of strategies that rewards can be collected from
+    mapping(address => bool) public supportedStrategies;
+    /// @notice Oracle contract used to validate swap prices
+    address public priceProvider;
+    /// @notice Maximum allowed slippage denominated in basis points. Example: 300 == 3% slippage
+    uint256 public allowedSlippageBps;
+    /// @notice Address receiving rewards proceeds. Initially this will be the ARM contract,
+    /// later this could be a Dripper contract that eases out rewards distribution.
+    address public rewardRecipient;
+
+    event SupportedStrategyUpdate(address strategy, bool isSupported);
+    event RewardTokenSwapped(
+        address indexed rewardToken,
+        address indexed swappedInto,
+        SwapPlatform swapPlatform,
+        uint256 amountIn,
+        uint256 amountOut
+    );
+    event RewardsCollected(address[] indexed strategy, address[][] rewardTokens, uint256[][] amounts);
+    event RewardRecipientUpdated(address rewardRecipient);
+    event AllowedSlippageUpdated(uint256 allowedSlippageBps);
+    event PriceProviderUpdated(address priceProvider);
+
+    error SlippageError(uint256 actualBalance, uint256 minExpected);
+    error BalanceMismatchAfterSwap(uint256 actualBalance, uint256 minExpected);
+    error InvalidSwapPlatform(SwapPlatform swapPlatform);
+    error UnsupportedStrategy(address strategyAddress);
+    error InvalidSwapRecipient(address recipient);
+    error InvalidFromAsset(address fromAsset);
+    error InvalidFromAssetAmount(uint256 fromAssetAmount);
+    error InvalidToAsset(address toAsset);
+    error EmptyLiquidityAsset();
+    error EmptyMagpieRouter();
+    error EmptyRewardRecipient();
+    error InvalidDecimals();
+    error InvalidAllowedSlippage(uint256 allowedSlippageBps);
+
+    constructor(address _liquidityAsset, address _magpieRouter) {
+        if (_liquidityAsset == address(0)) revert EmptyLiquidityAsset();
+        if (_magpieRouter == address(0)) revert EmptyMagpieRouter();
+        if (IERC20Metadata(_liquidityAsset).decimals() != 18) revert InvalidDecimals();
+
+        liquidityAsset = _liquidityAsset;
+        magpieRouter = _magpieRouter;
+    }
+
+    /// @notice
+    function initialize(address _priceProvider, uint256 _allowedSlippageBps, address _rewardRecipient)
+        external
+        initializer
```

Comment:
I like it slightly better when initialize has onlyOwner on it. This matters less now that we usually initialize during the same transaction as the proxy upgrade.

Reply:
I'll add `onlyOwner` to `initialize`